diff -cr typhoon-1.11.0.ORG/Makefile typhoon-1.11.0/Makefile
*** typhoon-1.11.0.ORG/Makefile	Sat Oct  2 03:37:32 1999
--- typhoon-1.11.0/Makefile	Mon Jan 20 01:53:31 2003
***************
*** 5,11 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MAKE		= make
  
--- 5,12 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MAKE		= make
  
diff -cr typhoon-1.11.0.ORG/configure typhoon-1.11.0/configure
*** typhoon-1.11.0.ORG/configure	Mon Oct  4 04:44:22 1999
--- typhoon-1.11.0/configure	Mon Jan 20 01:53:31 2003
***************
*** 30,36 ****
  compile='$CC $CARGS -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS=''	# additional -D compiler defines go here
  ENVIRON_H=./include/environ.h
  ANSI_H=./include/ansi.h
  
--- 30,36 ----
  compile='$CC $CARGS -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS='-DEXTEND'	# additional -D compiler defines go here
  ENVIRON_H=./include/environ.h
  ANSI_H=./include/ansi.h
  
Only in typhoon-1.11.0/examples: Makefile
diff -cr typhoon-1.11.0.ORG/examples/Makefile.in typhoon-1.11.0/examples/Makefile.in
*** typhoon-1.11.0.ORG/examples/Makefile.in	Sun Oct  3 23:28:28 1999
--- typhoon-1.11.0/examples/Makefile.in	Mon Jan 20 01:53:31 2003
***************
*** 8,14 ****
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
--- 8,15 ----
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
Only in typhoon-1.11.0/include: ansi.h
Only in typhoon-1.11.0/include: environ.h
diff -cr typhoon-1.11.0.ORG/man/Makefile typhoon-1.11.0/man/Makefile
*** typhoon-1.11.0.ORG/man/Makefile	Sat Oct  2 03:38:13 1999
--- typhoon-1.11.0/man/Makefile	Mon Jan 20 01:53:31 2003
***************
*** 7,13 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
--- 7,14 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
***************
*** 38,55 ****
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! 		-for manpage in $(MANPAGES); do \
! 			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! :			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! 			compress -c $$basepage.cat \
! 				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! 			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 		done
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
--- 39,58 ----
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! #		-for manpage in $(MANPAGES); do \
! #			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! #:			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! #			compress -c $$basepage.cat \
! #				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! #			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #		done
! 	cp d_*.3 $(PREFIX)/man/man3/
! 	cp ddlp.1 $(PREFIX)/man/man1/
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
Only in typhoon-1.11.0/src: Makefile
diff -cr typhoon-1.11.0.ORG/src/Makefile.in typhoon-1.11.0/src/Makefile.in
*** typhoon-1.11.0.ORG/src/Makefile.in	Sun Oct  3 23:28:28 1999
--- typhoon-1.11.0/src/Makefile.in	Mon Jan 20 01:53:31 2003
***************
*** 8,14 ****
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
--- 8,15 ----
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
diff -cr typhoon-1.11.0.ORG/src/unix.c typhoon-1.11.0/src/unix.c
*** typhoon-1.11.0.ORG/src/unix.c	Mon Oct  4 04:39:42 1999
--- typhoon-1.11.0/src/unix.c	Mon Jan 20 01:55:20 2003
***************
*** 152,158 ****
  
      while (fcntl(lock_fh, F_SETLK, &flk) == -1)
  	{
! 		if (errno != EINTR)
  		{
  			printf("ty_lock failed (errno %d, lock_fh %d)\n", errno, lock_fh);
  			break;
--- 152,158 ----
  
      while (fcntl(lock_fh, F_SETLK, &flk) == -1)
  	{
! 		if (errno != EINTR && errno != EAGAIN )
  		{
  			printf("ty_lock failed (errno %d, lock_fh %d)\n", errno, lock_fh);
  			break;
Only in typhoon-1.11.0/util: Makefile
diff -cr typhoon-1.11.0.ORG/util/Makefile.in typhoon-1.11.0/util/Makefile.in
*** typhoon-1.11.0.ORG/util/Makefile.in	Sun Oct  3 23:28:29 1999
--- typhoon-1.11.0/util/Makefile.in	Mon Jan 20 01:53:31 2003
***************
*** 10,16 ****
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c ddl.h exp.h imp.h
--- 10,17 ----
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c ddl.h exp.h imp.h
***************
*** 72,78 ****
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! 		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
--- 73,79 ----
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! #		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
diff -cr typhoon-1.11.0.ORG/util/export.c typhoon-1.11.0/util/export.c
*** typhoon-1.11.0.ORG/util/export.c	Mon Oct  4 04:11:31 1999
--- typhoon-1.11.0/util/export.c	Mon Jan 20 01:53:31 2003
***************
*** 74,85 ****
--- 74,99 ----
  FILE *lex_file;
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyexport [option]... database[.dbd]\n\
  Options:\n\
+     -s          export to stdout (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate export specification\n\
      -n          Strings are not null-terminated\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyexport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate export specification\n\
+     -n          Strings are not null-terminated\n";
+ #endif
+ 
+ #ifdef EXTEND
+ int OutfileIsStdout = 0;
+ #endif
  
  #ifdef CONFIG_PROTOTYPES
  void err_quit(char *s, ...)
***************
*** 335,344 ****
--- 349,369 ----
  		{
  			sprintf(export_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (OutfileIsStdout == 1){
+ 				outfile = stdout;
+ 			}else{
+ 				if( !(outfile = fopen(export_fname, "w")) )
+ 					err_quit("Cannot write to '%s'", export_fname);
+ 
+ 				printf("exporting to '%s'\n", export_fname);
+ 			}
+ #else
  		 	if( !(outfile = fopen(export_fname, "w")) )
  				err_quit("Cannot write to '%s'", export_fname);
  
  			printf("exporting to '%s'\n", export_fname);
+ #endif
  			ExportTable(i);
  			fclose(outfile);
  		}
***************
*** 398,403 ****
--- 423,433 ----
  		{
  			switch( argv[i][1] )
  			{
+ #ifdef EXTEND
+ 				case 's':
+ 					OutfileIsStdout = 1;
+ 					break;
+ #endif
  				case 'f':
  					d_dbfpath(argv[i]+2);
  					break;
diff -cr typhoon-1.11.0.ORG/util/import.c typhoon-1.11.0/util/import.c
*** typhoon-1.11.0.ORG/util/import.c	Mon Oct  4 04:11:32 1999
--- typhoon-1.11.0/util/import.c	Mon Jan 20 01:53:31 2003
***************
*** 79,89 ****
--- 79,102 ----
  static char		import_fname[256];
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyimport [option]... database[.dbd]\n\
  Options:\n\
+     -s          import from stdin (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate import specification\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyimport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate import specification\n";
+ #endif
+ 
+ #ifdef EXTEND
+ static int            InfileIsStdin = 0;
+ #endif
  
  #ifdef CONFIG_PROTOTYPES
  void err_quit(char *s, ...)
***************
*** 498,507 ****
--- 511,531 ----
  			lineno = 1;
  			sprintf(import_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (InfileIsStdin == 1){
+ 				infile = stdin;
+ 			}else{
+ 				if( !(infile = fopen(import_fname, "r")) )
+ 					err_quit("Cannot open '%s'", import_fname);
+ 
+ 				printf("importing from '%s'\n", import_fname);
+ }
+ #else
  			if( !(infile = fopen(import_fname, "r")) )
  				err_quit("Cannot open '%s'", import_fname);
  
  			printf("importing from '%s'\n", import_fname);
+ #endif
  			ImportTable(i);
  			fclose(infile);
  		}
***************
*** 568,573 ****
--- 592,603 ----
  				case 'g':
  					GenerateImportSpec(realname);
  					exit(1);
+ #ifdef EXTEND
+ 				case 's':
+ 					InfileIsStdin = 1;
+ 					break;
+ #endif
+ 
  				default:
  					err_quit("unknown command line option");
  			}
