diff -c -r typhoon-1.10.3.ORG/Makefile typhoon-1.10.3/Makefile
*** typhoon-1.10.3.ORG/Makefile	Mon May 22 01:47:05 1995
--- typhoon-1.10.3/Makefile	Tue Sep  7 00:37:13 1999
***************
*** 5,11 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MAKE		= make
  
--- 5,12 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MAKE		= make
  
diff -c -r typhoon-1.10.3.ORG/configure typhoon-1.10.3/configure
*** typhoon-1.10.3.ORG/configure	Mon May 22 01:47:06 1995
--- typhoon-1.10.3/configure	Thu Feb  3 11:15:44 2000
***************
*** 6,12 ****
  compile='$CC -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS=''
  ANSI_OBJS=''
  ANSI_SRCS=''
  ENVIRON_H=./include/environ.h
--- 6,12 ----
  compile='$CC -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS='-DEXTEND'
  ANSI_OBJS=''
  ANSI_SRCS=''
  ENVIRON_H=./include/environ.h
Only in typhoon-1.10.3: data
diff -c -r typhoon-1.10.3.ORG/examples/Makefile.in typhoon-1.10.3/examples/Makefile.in
*** typhoon-1.10.3.ORG/examples/Makefile.in	Mon May 22 01:47:11 1995
--- typhoon-1.10.3/examples/Makefile.in	Tue Sep  7 00:36:37 1999
***************
*** 10,16 ****
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
--- 10,17 ----
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
diff -c -r typhoon-1.10.3.ORG/man/Makefile typhoon-1.10.3/man/Makefile
*** typhoon-1.10.3.ORG/man/Makefile	Tue Jun 13 03:58:08 1995
--- typhoon-1.10.3/man/Makefile	Tue Sep  7 00:52:25 1999
***************
*** 7,13 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
--- 7,14 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
***************
*** 38,55 ****
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! 		-for manpage in $(MANPAGES); do \
! 			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! :			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! 			compress -c $$basepage.cat \
! 				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! 			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 		done
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
--- 39,58 ----
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! #		-for manpage in $(MANPAGES); do \
! #			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! #:			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! #			compress -c $$basepage.cat \
! #				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! #			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #		done
! 	cp d_*.3 $(PREFIX)/man/man3/
! 	cp ddlp.1 $(PREFIX)/man/man1/
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
diff -c -r typhoon-1.10.3.ORG/src/Makefile.in typhoon-1.10.3/src/Makefile.in
*** typhoon-1.10.3.ORG/src/Makefile.in	Mon May 22 01:47:27 1995
--- typhoon-1.10.3/src/Makefile.in	Tue Sep  7 00:37:38 1999
***************
*** 10,16 ****
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
--- 10,17 ----
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
diff -c -r typhoon-1.10.3.ORG/util/Makefile.in typhoon-1.10.3/util/Makefile.in
*** typhoon-1.10.3.ORG/util/Makefile.in	Mon May 22 01:47:41 1995
--- typhoon-1.10.3/util/Makefile.in	Tue Sep  7 00:40:20 1999
***************
*** 12,18 ****
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c
--- 12,19 ----
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c
***************
*** 74,80 ****
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! 		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
--- 75,81 ----
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! #		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
diff -c -r typhoon-1.10.3.ORG/util/export.c typhoon-1.10.3/util/export.c
*** typhoon-1.10.3.ORG/util/export.c	Mon May 22 01:47:45 1995
--- typhoon-1.10.3/util/export.c	Thu Feb  3 11:23:34 2000
***************
*** 68,80 ****
--- 68,93 ----
  FILE *lex_file;
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyexport [option]... database[.dbd]\n\
  Options:\n\
+     -s          export to stdout (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate export specification\n\
      -n          Strings are not null-terminated\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyexport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate export specification\n\
+     -n          Strings are not null-terminated\n";
+ #endif
  
+ #ifdef EXTEND
+ int OutfileIsStdout = 0;
+ #endif
  
  
  istrcmp(s1, s2)
***************
*** 343,352 ****
--- 356,376 ----
  		{
  			sprintf(export_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (OutfileIsStdout == 1){
+ 				outfile = stdout;
+ 			}else{
+ 				if( !(outfile = fopen(export_fname, "w")) )
+ 					err_quit("Cannot write to '%s'", export_fname);
+ 
+ 				printf("exporting to '%s'\n", export_fname);
+ 			}
+ #else
  		 	if( !(outfile = fopen(export_fname, "w")) )
  				err_quit("Cannot write to '%s'", export_fname);
  
  			printf("exporting to '%s'\n", export_fname);
+ #endif
  			ExportTable(i);
  			fclose(outfile);
  		}
***************
*** 406,411 ****
--- 430,440 ----
  		{
  			switch( argv[i][1] )
  			{
+ #ifdef EXTEND
+ 				case 's':
+ 					OutfileIsStdout = 1;
+ 					break;
+ #endif
  				case 'f':
  					d_dbfpath(argv[i]+2);
  					break;
diff -c -r typhoon-1.10.3.ORG/util/expspec.c typhoon-1.10.3/util/expspec.c
*** typhoon-1.10.3.ORG/util/expspec.c	Mon May 22 01:47:47 1995
--- typhoon-1.10.3/util/expspec.c	Fri Feb 25 13:22:27 2000
***************
*** 73,79 ****
--- 73,83 ----
  
  	yyparse();
  
+ #if 0
      fclose(outfile);
+ #else
+     fclose(lex_file);
+ #endif
  }
  
  
diff -c -r typhoon-1.10.3.ORG/util/import.c typhoon-1.10.3/util/import.c
*** typhoon-1.10.3.ORG/util/import.c	Mon May 22 01:47:48 1995
--- typhoon-1.10.3/util/import.c	Fri Feb  4 12:42:23 2000
***************
*** 73,84 ****
--- 73,96 ----
  static char		import_fname[256];
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyimport [option]... database[.dbd]\n\
  Options:\n\
+     -s          import from stdin (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate import specification\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyimport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate import specification\n";
+ #endif
  
+ #ifdef EXTEND
+ static int            InfileIsStdin = 0;
+ #endif
  
  
  istrcmp(s1, s2)
***************
*** 504,513 ****
--- 516,536 ----
  			lineno = 1;
  			sprintf(import_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (InfileIsStdin == 1){
+ 				infile = stdin;
+ 			}else{
+ 				if( !(infile = fopen(import_fname, "r")) )
+ 					err_quit("Cannot open '%s'", import_fname);
+ 
+ 				printf("importing from '%s'\n", import_fname);
+ }
+ #else
  			if( !(infile = fopen(import_fname, "r")) )
  				err_quit("Cannot open '%s'", import_fname);
  
  			printf("importing from '%s'\n", import_fname);
+ #endif
  			ImportTable(i);
  			fclose(infile);
  		}
***************
*** 574,579 ****
--- 597,608 ----
  				case 'g':
  					GenerateImportSpec(realname);
  					exit(1);
+ #ifdef EXTEND
+ 				case 's':
+ 					InfileIsStdin = 1;
+ 					break;
+ #endif
+ 
  				default:
  					err_quit("unknown command line option");
  			}
diff -c -r typhoon-1.10.3.ORG/util/impspec.c typhoon-1.10.3/util/impspec.c
*** typhoon-1.10.3.ORG/util/impspec.c	Mon May 22 01:47:49 1995
--- typhoon-1.10.3/util/impspec.c	Fri Feb  4 12:41:32 2000
***************
*** 71,77 ****
--- 71,81 ----
  
  	yyparse();
  
+ #if 0
      fclose(outfile);
+ #else
+     fclose(lex_file);
+ #endif
  }
  
  
