diff -cr typhoon-1.11.0.ORG/Makefile typhoon-1.11.0/Makefile
*** typhoon-1.11.0.ORG/Makefile	Sat Oct  2 12:37:32 1999
--- typhoon-1.11.0/Makefile	Thu Feb 20 12:10:05 2003
***************
*** 5,11 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MAKE		= make
  
--- 5,12 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MAKE		= make
  
Only in typhoon-1.11.0: Makefile.orig
diff -cr typhoon-1.11.0.ORG/configure typhoon-1.11.0/configure
*** typhoon-1.11.0.ORG/configure	Mon Oct  4 13:44:22 1999
--- typhoon-1.11.0/configure	Thu Feb 20 12:29:24 2003
***************
*** 15,21 ****
  # typhoon uses a mixture of POSIX, BSD and System V interfaces
  #
  
! STANDARDS="-D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE"
  
  if type gcc >/dev/null 2>/dev/null ; then
  	CC=gcc
--- 15,22 ----
  # typhoon uses a mixture of POSIX, BSD and System V interfaces
  #
  
! #STANDARDS="-D_POSIX_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE"
! STANDARDS="-D_BSD_SOURCE -D_SVID_SOURCE"
  
  if type gcc >/dev/null 2>/dev/null ; then
  	CC=gcc
***************
*** 30,36 ****
  compile='$CC $CARGS -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS=''	# additional -D compiler defines go here
  ENVIRON_H=./include/environ.h
  ANSI_H=./include/ansi.h
  
--- 31,37 ----
  compile='$CC $CARGS -o conftest conftest.c >/dev/null 2>&1'
  compiled_ok='test -s conftest && (./conftest) >/dev/null 2>/dev/null;'
  
! DEFS='-DEXTEND'	# additional -D compiler defines go here
  ENVIRON_H=./include/environ.h
  ANSI_H=./include/ansi.h
  
Only in typhoon-1.11.0: configure.orig
Only in typhoon-1.11.0/examples: Makefile
diff -cr typhoon-1.11.0.ORG/examples/Makefile.in typhoon-1.11.0/examples/Makefile.in
*** typhoon-1.11.0.ORG/examples/Makefile.in	Mon Oct  4 08:28:28 1999
--- typhoon-1.11.0/examples/Makefile.in	Thu Feb 20 12:10:06 2003
***************
*** 3,14 ****
  DEFINES		= -I../include @defs@
  CC		= @cc@
  CFLAGS		= @cflags@
! LIBS		= -ltyphoon
  PREFIX		= /usr/local
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
--- 3,15 ----
  DEFINES		= -I../include @defs@
  CC		= @cc@
  CFLAGS		= @cflags@
! LIBS		= -ltyphoon -lcompat
  PREFIX		= /usr/local
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= wheel
  SHELL		= /bin/sh
  PROGRAM		= demo
  SRCS		= demo.c
Only in typhoon-1.11.0/examples: Makefile.in.orig
Only in typhoon-1.11.0/examples: demo
Only in typhoon-1.11.0/examples: demo.dbd
Only in typhoon-1.11.0/examples: demo.h
Only in typhoon-1.11.0/examples: demo.o
Only in typhoon-1.11.0/include: ansi.h
Only in typhoon-1.11.0/include: environ.h
diff -cr typhoon-1.11.0.ORG/man/Makefile typhoon-1.11.0/man/Makefile
*** typhoon-1.11.0.ORG/man/Makefile	Sat Oct  2 12:38:13 1999
--- typhoon-1.11.0/man/Makefile	Thu Feb 20 12:10:06 2003
***************
*** 7,13 ****
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
--- 7,14 ----
  DESTMAN		= $(PREFIX)/man/man$(MANEXT)
  DESTCAT		= $(PREFIX)/man/cat$(MANEXT)
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= root
  SHELL		= /bin/sh
  MANPAGES	= d_close.3 d_crget.3 d_crread.3 d_crset.3 d_dbdpath.3 \
  		  d_dbfpath.3 d_dbget.3 d_dbset.3 d_delete.3 d_fillnew.3 \
***************
*** 38,55 ****
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! 		-for manpage in $(MANPAGES); do \
! 			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! :			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! :			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! 			compress -c $$basepage.cat \
! 				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! 			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! 		done
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
--- 39,58 ----
  $(CATPAGES):	$(MANPAGES)
  
  install:	$(CATPAGES)
! #		-for manpage in $(MANPAGES); do \
! #			basepage=`expr $$manpage : '\(.*\)\.[^.]*'`; \
! #:			cp $$manpage $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chmod 644 $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chown $(DESTOWN) $(DESTMAN)/$$basepage.$(MANEXT); \
! #:			chgrp $(DESTGRP) $(DESTMAN)/$$basepage.$(MANEXT); \
! #			compress -c $$basepage.cat \
! #				>$(DESTCAT)/$$basepage.$(MANEXT).Z; \
! #			chmod 644 $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chown $(DESTOWN) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #			chgrp $(DESTGRP) $(DESTCAT)/$$basepage.$(MANEXT)*; \
! #		done
! 	cp d_*.3 $(PREFIX)/man/man3/
! 	cp ddlp.1 $(PREFIX)/man/man1/
  
  uninstall:
  		-for manpage in $(MANPAGES); do \
Only in typhoon-1.11.0/man: Makefile.orig
Only in typhoon-1.11.0/src: Makefile
diff -cr typhoon-1.11.0.ORG/src/Makefile.in typhoon-1.11.0/src/Makefile.in
*** typhoon-1.11.0.ORG/src/Makefile.in	Mon Oct  4 08:28:28 1999
--- typhoon-1.11.0/src/Makefile.in	Thu Feb 20 12:10:06 2003
***************
*** 8,14 ****
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
--- 8,15 ----
  DESTLIB		= $(PREFIX)/lib
  DESTHDR		= $(PREFIX)/include
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= wheel
  SHELL		= /bin/sh
  LIBRARY		= libtyphoon.a
  LIBHDRS		= ../include/environ.h ../include/typhoon.h
Only in typhoon-1.11.0/src: Makefile.in.orig
Only in typhoon-1.11.0/src: ansi.o
Only in typhoon-1.11.0/src: bt_del.o
Only in typhoon-1.11.0/src: bt_funcs.o
Only in typhoon-1.11.0/src: bt_io.o
Only in typhoon-1.11.0/src: bt_open.o
Only in typhoon-1.11.0/src: cmpfuncs.o
Only in typhoon-1.11.0/src: libtyphoon.a
diff -cr typhoon-1.11.0.ORG/src/os.c typhoon-1.11.0/src/os.c
*** typhoon-1.11.0.ORG/src/os.c	Mon Oct  4 12:45:07 1999
--- typhoon-1.11.0/src/os.c	Thu Feb 20 12:10:06 2003
***************
*** 35,40 ****
--- 35,41 ----
   *
   *--------------------------------------------------------------------------*/
  
+ #    include <sys/file.h>
  #include "environ.h"
  #ifdef CONFIG_UNIX
  #  include <sys/types.h>
***************
*** 138,143 ****
--- 139,147 ----
  long offset;
  unsigned bytes;
  {
+ #ifdef CONFIG_USE_FLOCK
+ 	struct flock flk;
+ #endif
  	lseek(fh, offset, SEEK_SET);
  #ifdef CONFIG_DOS
  	/* Microsoft C */
***************
*** 146,152 ****
--- 150,165 ----
  #	endif
  #endif
  #ifdef CONFIG_UNIX
+ # ifdef CONFIG_USE_FLOCK
+ 	flk.l_type = F_UNLCK;
+ 	flk.l_whence = SEEK_SET;
+ 	flk.l_start = 0;
+ 	flk.l_len = bytes;
+ 
+ 	return fcntl(fh, F_SETLK, &flk);
+ # else
  	return lockf(fh, F_ULOCK, bytes);
+ # endif
  #endif
  #ifdef CONFIG_OS2
  #	ifdef __BORLANDC__
Only in typhoon-1.11.0/src: os.c.orig
Only in typhoon-1.11.0/src: os.o
Only in typhoon-1.11.0/src: readdbd.o
Only in typhoon-1.11.0/src: record.o
Only in typhoon-1.11.0/src: sequence.o
Only in typhoon-1.11.0/src: ty_auxfn.o
Only in typhoon-1.11.0/src: ty_find.o
Only in typhoon-1.11.0/src: ty_ins.o
Only in typhoon-1.11.0/src: ty_io.o
Only in typhoon-1.11.0/src: ty_log.o
Only in typhoon-1.11.0/src: ty_open.o
Only in typhoon-1.11.0/src: ty_refin.o
Only in typhoon-1.11.0/src: ty_repl.o
Only in typhoon-1.11.0/src: ty_util.o
diff -cr typhoon-1.11.0.ORG/src/unix.c typhoon-1.11.0/src/unix.c
*** typhoon-1.11.0.ORG/src/unix.c	Mon Oct  4 13:39:42 1999
--- typhoon-1.11.0/src/unix.c	Thu Feb 20 12:26:31 2003
***************
*** 152,158 ****
  
      while (fcntl(lock_fh, F_SETLK, &flk) == -1)
  	{
! 		if (errno != EINTR)
  		{
  			printf("ty_lock failed (errno %d, lock_fh %d)\n", errno, lock_fh);
  			break;
--- 152,158 ----
  
      while (fcntl(lock_fh, F_SETLK, &flk) == -1)
  	{
! 		if( errno != EINTR && errno != EAGAIN )
  		{
  			printf("ty_lock failed (errno %d, lock_fh %d)\n", errno, lock_fh);
  			break;
***************
*** 187,195 ****
--- 187,208 ----
  #endif
  
  #ifndef SEMLOCK
+ #ifdef CONFIG_USE_FLOCK
+         struct flock flk;
+ #endif
+ 
  	lseek(lock_fh, 0, SEEK_SET);
  
+ #ifdef CONFIG_USE_FLOCK
+         flk.l_type = F_UNLCK;
+         flk.l_whence = SEEK_SET;
+         flk.l_start = 0;
+         flk.l_len = 1;
+ 
+ 	while( fcntl(lock_fh, F_SETLK, &flk) == -1 )
+ #else
  	while( lockf(lock_fh, F_ULOCK, 1) == -1 )
+ #endif
  	{
  		if( errno != EINTR && errno != EAGAIN )
  		{
Only in typhoon-1.11.0/src: unix.c.orig
Only in typhoon-1.11.0/src: unix.o
Only in typhoon-1.11.0/src: vlr.o
Only in typhoon-1.11.0/util: Makefile
diff -cr typhoon-1.11.0.ORG/util/Makefile.in typhoon-1.11.0/util/Makefile.in
*** typhoon-1.11.0.ORG/util/Makefile.in	Mon Oct  4 08:28:29 1999
--- typhoon-1.11.0/util/Makefile.in	Thu Feb 20 12:10:07 2003
***************
*** 5,16 ****
  YFLAGS		= -d
  CC		= @cc@
  CFLAGS		= @cflags@
! LIBS		= -ltyphoon
  PREFIX		= /usr/local
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! DESTGRP		= local
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c ddl.h exp.h imp.h
--- 5,17 ----
  YFLAGS		= -d
  CC		= @cc@
  CFLAGS		= @cflags@
! LIBS		= -ltyphoon -lcompat
  PREFIX		= /usr/local
  LDFLAGS		= -L../src
  DESTBIN		= $(PREFIX)/bin
  DESTOWN		= root
! #DESTGRP		= local
! DESTGRP		= wheel
  SHELL		= /bin/sh
  PROGRAMS	= ddlp dbdview tyexport tyimport # tybackup tyrestore
  MADESRCS	= ddl.c exp.c imp.c ddl.h exp.h imp.h
***************
*** 72,78 ****
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! 		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
--- 73,79 ----
  
  install:	$(PROGRAMS)
  		cp $(PROGRAMS) $(DESTBIN)
! #		cd $(DESTBIN) && -mcs -c $(PROGRAMS)
  		cd $(DESTBIN) && strip $(PROGRAMS)
  		cd $(DESTBIN) && chmod 755 $(PROGRAMS)
  		cd $(DESTBIN) && chown $(DESTOWN) $(PROGRAMS)
Only in typhoon-1.11.0/util: Makefile.in.orig
Only in typhoon-1.11.0/util: dbdview
Only in typhoon-1.11.0/util: dbdview.o
Only in typhoon-1.11.0/util: ddl.c
Only in typhoon-1.11.0/util: ddl.h
Only in typhoon-1.11.0/util: ddl.o
Only in typhoon-1.11.0/util: ddlp
Only in typhoon-1.11.0/util: ddlp.o
Only in typhoon-1.11.0/util: ddlplex.o
Only in typhoon-1.11.0/util: ddlpsym.o
Only in typhoon-1.11.0/util: exp.c
Only in typhoon-1.11.0/util: exp.h
Only in typhoon-1.11.0/util: exp.o
diff -cr typhoon-1.11.0.ORG/util/export.c typhoon-1.11.0/util/export.c
*** typhoon-1.11.0.ORG/util/export.c	Mon Oct  4 13:11:31 1999
--- typhoon-1.11.0/util/export.c	Thu Feb 20 12:10:07 2003
***************
*** 74,85 ****
--- 74,99 ----
  FILE *lex_file;
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyexport [option]... database[.dbd]\n\
  Options:\n\
+     -s          export to stdout (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate export specification\n\
      -n          Strings are not null-terminated\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyexport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate export specification\n\
+     -n          Strings are not null-terminated\n";
+ #endif
+ 
+ #ifdef EXTEND
+ int OutfileIsStdout = 0;
+ #endif
  
  #ifdef CONFIG_PROTOTYPES
  void err_quit(char *s, ...)
***************
*** 335,344 ****
--- 349,369 ----
  		{
  			sprintf(export_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (OutfileIsStdout == 1){
+ 				outfile = stdout;
+ 			}else{
+ 				if( !(outfile = fopen(export_fname, "w")) )
+ 					err_quit("Cannot write to '%s'", export_fname);
+ 
+ 				printf("exporting to '%s'\n", export_fname);
+ 			}
+ #else
  		 	if( !(outfile = fopen(export_fname, "w")) )
  				err_quit("Cannot write to '%s'", export_fname);
  
  			printf("exporting to '%s'\n", export_fname);
+ #endif
  			ExportTable(i);
  			fclose(outfile);
  		}
***************
*** 398,403 ****
--- 423,433 ----
  		{
  			switch( argv[i][1] )
  			{
+ #ifdef EXTEND
+ 				case 's':
+ 					OutfileIsStdout = 1;
+ 					break;
+ #endif
  				case 'f':
  					d_dbfpath(argv[i]+2);
  					break;
Only in typhoon-1.11.0/util: export.c.orig
Only in typhoon-1.11.0/util: export.o
Only in typhoon-1.11.0/util: exportlx.o
Only in typhoon-1.11.0/util: expspec.o
Only in typhoon-1.11.0/util: imp.c
Only in typhoon-1.11.0/util: imp.h
Only in typhoon-1.11.0/util: imp.o
diff -cr typhoon-1.11.0.ORG/util/import.c typhoon-1.11.0/util/import.c
*** typhoon-1.11.0.ORG/util/import.c	Mon Oct  4 13:11:32 1999
--- typhoon-1.11.0/util/import.c	Thu Feb 20 12:10:07 2003
***************
*** 79,89 ****
--- 79,102 ----
  static char		import_fname[256];
  
  /*------------------------------ local variables ---------------------------*/
+ #ifdef EXTEND
  static char paramhelp[] = "\
  Syntax: tyimport [option]... database[.dbd]\n\
  Options:\n\
+     -s          import from stdin (extension by M.Ito)\n\
      -f<path>    Specify data files path\n\
      -g          Generate import specification\n";
+ #else
+ static char paramhelp[] = "\
+ Syntax: tyimport [option]... database[.dbd]\n\
+ Options:\n\
+     -f<path>    Specify data files path\n\
+     -g          Generate import specification\n";
+ #endif
+ 
+ #ifdef EXTEND
+ static int            InfileIsStdin = 0;
+ #endif
  
  #ifdef CONFIG_PROTOTYPES
  void err_quit(char *s, ...)
***************
*** 498,507 ****
--- 511,531 ----
  			lineno = 1;
  			sprintf(import_fname, "%s.kom", dbd.record[i].name);
  
+ #ifdef EXTEND
+ 			if (InfileIsStdin == 1){
+ 				infile = stdin;
+ 			}else{
+ 				if( !(infile = fopen(import_fname, "r")) )
+ 					err_quit("Cannot open '%s'", import_fname);
+ 
+ 				printf("importing from '%s'\n", import_fname);
+ }
+ #else
  			if( !(infile = fopen(import_fname, "r")) )
  				err_quit("Cannot open '%s'", import_fname);
  
  			printf("importing from '%s'\n", import_fname);
+ #endif
  			ImportTable(i);
  			fclose(infile);
  		}
***************
*** 568,573 ****
--- 592,603 ----
  				case 'g':
  					GenerateImportSpec(realname);
  					exit(1);
+ #ifdef EXTEND
+ 				case 's':
+ 					InfileIsStdin = 1;
+ 					break;
+ #endif
+ 
  				default:
  					err_quit("unknown command line option");
  			}
Only in typhoon-1.11.0/util: import.c.orig
Only in typhoon-1.11.0/util: import.o
Only in typhoon-1.11.0/util: importlx.o
Only in typhoon-1.11.0/util: impspec.o
Only in typhoon-1.11.0/util: tyexport
Only in typhoon-1.11.0/util: tyimport
